src/main/java/server/Client.java

```diff
@@ -21,10 +21,14 @@ public Client(Socket clientSocket) throws IOException {
     public PrintWriter getWriter() { return writer; }
 
     public String chunkStream() {
-        char[] charBuffer = new char[100000];
+        char[] charBuffer = new char[1000];
```

Comment:
What's the effect of making this number smaller?

Reply:
I'm pretty sure that with the higher buffer `chunkStream` would hit the end of the incoming stream, and for `BufferedReader.read()` hitting the end of stream returns -1, which is what was causing that `indexOutOfBounds` exception to occur.